### ISSUE TITLE
Order of discards of globally known trash still unclear with UDD/UTDs

### ISSUE THREAD

**Issue by english5040** [hanabi/hanabi.github.io#1214]
It's still unclear in what order you should discard the focus of a UDD vs. other known trash to its left. The conventions say you should discard the focus "first", but only among a UDD touching 2 cards; going by that, I would think you also discard the focus of a UDD before other trash cards, but it's unclear in the docs.

It would make sense to discard the focus of a possible UDD/UTD first in all cases, since discarding can give more information on what card was duplicated if the clue was a UDD.

(May make a PR later if I have time)

**Comment by Zamiell** [comment]
you should discard the focus first if the identity of the card is not yet known. otherwise, you discard the left-most.
feel free to submit a pr

**Comment by english5040** [comment]
Another special case: even if only one new card was introduced in the UTD/UDD, discarding it first is still beneficial because of obtaining elimination notes for the possibly-duplicated card. The UTD/UDD would need to be a clue that only has one card it applies to in the hand. Maybe it's simpler if we do focus-first discarding in all cases?

I guess there's also the case where you discard the unfocused card of a UDD with known trash available, too.

**Comment by Zamiell** [comment]
> Maybe it's simpler if we do focus-first discarding in all cases?

well, consider that:
1) the general case of the clue receiver is that they don't know whether or not the clue is a UTD or a UDD yet.
2) in 80%~ of cases, it is a UTD.

so forcing discarding of focus first with the intent to improve UDD ends up affecting a ton of UTD situations.
so it seems probably undesirable

**Comment by Zamiell** [comment]
@english5040 are you going to do a PR or should I close this?

**Comment by english5040** [comment]
Recently been busy thinking about other convention systems, sorry.

Currently I am still thinking about how best to do things, I don't like changing the discard order too much. I do think UDDs are, anecdotally, very likely- a large percentage of discharges are UDDs (happens in early game a lot), or appear likely to be UDDs to the receiver. Anyone have ideas?

**Comment by Zamiell** [comment]
Yeah I don't want to change discard order either, but did you want to do a a purely-clarification pr ? Since in the OP you said the doc was confusing as-currently-written.

**Comment by Zamiell** [comment]
> if a UDD happens (lets say on 2 cards) and then both cards happen to be fully known, what is the expected discard?

i dont understand the question because one card is good and one card is bad, so the expected discarded is only the bad card

> A UDD/UTD and another trash move happens. For example Alice's hand is [kt] x [r] [r] and red was used as UTD. Which is the expected discard?

from the clue receiver's perspective:
- if it CAN be a UDD, then they need to discard slot 4 first in order to gain information about their hand (i.e. the identity of slot 3).
- if it CANT be a UDD, then slot 1 and slot 4 are both known trash, so you discard trash from left to right, so yo discard slot 1

i agree that this is stupid/confusing but this does not happen often, and there are other tradeoffs involved with changing anything

**Comment by ricardodd2** [comment]
> * if it CAN be a UDD, then they need to discard slot 4 first in order to gain information about their hand (i.e. the identity of slot 3).
Is there written in the doc that that a potential UDD is the expected discard over another kt on the left? (This should imply that discarding kt on left should trash order cm)
If not, I will suggest a simple definition for potential UDD as "a UTD or UDD that touched more than 1 card". This seems clear enough that everyone should agree which trash has precedence and should not harm even if a UTD touching multiple cards is globally known not to be UDD (maybe specify thia in the doc if we are going to add it: that even a globally known UTD that touched multiple cards is considered a "potential UDD" for simplicity)

**Comment by Zamiell** [comment]
> (This should imply that discarding kt on left should trash order cm)

not really, i think it implies that it is undefined and should be treated as a known mistake

> just add "(even before other known trash on the left of it)"

sounds like a simple enough clarification, do you want to do a PR?

**Comment by ricardodd2** [comment]
> > just add "(even before other known trash on the left of it)"
> 
> sounds like a simple enough clarification, do you want to do a PR?

Ok

**Comment by Zamiell** [comment]
@english5040 i will close this issue for now after #1231, we can open another issue/PR if needed
